anomaly-benchmarking/generate-datasets/script.whizzml

```diff
@@ -0,0 +1,205 @@
+;; Takes a dataset, returns the same dataset with all fields marked preferred
+
+(define (all-preferred ds-id)
+  (let (ds (fetch ds-id)
+        field-list (keys (ds "fields"))
+        not-preferred (filter (lambda (x) (not (get-in ds ["fields" x "preferred"]))) field-list))
+     (update ds-id {"fields" (make-map not-preferred (repeat (count not-preferred) {"preferred" true}))})))
+
+;; Given a dataset and a field name, returns the field id with that name
+
+(define (get-id ds name)
+  (let (field (ds "fields"))
+    (head (filter (lambda (x) (= (field [x "name"]) name)) 
+                (keys field)))))
+
+;; Given a binary dataset, finds the point difficulty of each row and
+;; returns an dataset of only anomalous points and a new field
+;; labeling each row by one of four difficulty levels.
+
+(define (point-difficulty ds-id delete-resources)
+  (let (ds (all-preferred ds-id)
+        lr-id (create-and-wait-logisticregression {"dataset" ds})
+        lr-fields ((fetch lr-id) "input_fields")
+        bp-id (create-and-wait-batchprediction {"logisticregression" lr-id 
+                                       "dataset" ds 
+                                       "probabilities" true 
+                                       "output_dataset" true 
+                                       "output_fields" lr-fields })
```

Comment:
yes, that's expected. in general, in a batch prediction the input
dataset won't contain the objective,  and if it does, its value doesn't
need to be the same as the predicted one, so having it twice makes sense
(one from the input, which is not used, one the predicted value).